File: API-Server/Happy Habits App/Repositories/IMessageRepository.cs
using Happy_Habits_App.Model;

namespace Happy_Habits_App.Repositories
{
    public interface IMessageRepository
    {
        Task<Queue<Message>> GetAllMessagesByGroupId(string groupId);

        Task UpdateFriendGroupChat(FriendGroup group);

        Task<FriendGroup> GetFriendGroupById(string groupId);

        Task<List<FriendGroup>> GetAllFriends(string userId);
        Task CreateFriendGroup(FriendGroup friendGroup);
        Task<FriendGroup> GetFriendGroupByScannerIdByGenId(string scannerId, string genId);
        Task<string> GetFriendIdByGroupIdAndSenderId(string groupId, string senderId);
    }
}

File: Mobile-App/app/src/main/java/com/example/happyhabits/feature_application/feature_chat/domain/repository/IMessageRepository.kt
package com.example.happyhabits.feature_application.feature_chat.domain.repository

import com.example.happyhabits.feature_application.feature_chat.domain.model.FriendGroup
import com.example.happyhabits.feature_application.feature_chat.domain.model.Message

interface IMessageRepository {
    suspend fun getMessages(groupId: String): List<Message>

    suspend fun getFriendList(userId: String): List<FriendGroup>
}